dashboard/lib/build_dashboard_page.dart

```diff
@@ -78,7 +79,13 @@ class BuildDashboardPageState extends State<BuildDashboardPage> {
       queryParameters.addAll(_filter!.toMap(includeDefaults: false));
     }
     queryParameters['repo'] = repo!;
-    queryParameters['branch'] = branch!;
+
+    Set<String> validOptions =
+        buildState.branches.where((Branch b) => b.repository == repo).map((Branch b) => b.branch).toSet();
+    branch = validOptions.contains(branch) ? branch : 'master';
+    queryParameters['branch'] = validOptions.contains(branch) ? branch! : defaultBranches[repo]!;
+    // FOR REVIEW: If we switch to a repo where the current release branch does not exist, I am updating the branch to be master
```

Comment:
Don't do this. Always assume the user knows more than the system.

For example, I can query directly https://flutter-dashboard.appspot.com/#/build?repo=engine&branch=flutter-2.13-candidate.0 but this change would break that

Reply:
umm my understanding is that branches flutter-2.13-candidate.0 will be reachable when the repository is valid, such as engine. Currently I am trying to prevent an access of branch from an invalid repository, such as from impeller repository. This happens when the branches dropdown stays the same, and when users toggles the repository dropdown. I am trying to use the current logic to help branches dropdown get updated when the repository dropdown is updated.

Reply:
It's a better UX to have this fail by showing an empty dashboard than to have it lock someone out.

For example, if this was in place last week I wouldn't have been able to drive the google3 CPs as the branch didn't exist according to flutter-dashboard

Reply:
Sure, I handled the transition state by keeping the latest branch selection for the user, along with all the valid branch options that would come from the newest repository selection.